### DISCUSSION TITLE
Flatbuffer LUT Calibration

### DISCUSSION THREAD

**Discussion by NatyaSadella** [awawa-dev/HyperHDR#467] [Q&A]
I need help with the Flat LUT Calibration in HyperHDR. I bought a new graphics card to be able to see the HyperHDR interface in HDR on chrome. That worked so I've tried to calibrate the HDR LUT File.

A) HyperHDR always says "Press F11 for Fullscreen-Mode". Fullscreen Mode is enabled BEFORE I click on Start so I don't know why I get that error. Firefox is the only browser to fix this.

B) I just get a mosaic like pixel picture on the TV screen, is it supposed to look like that? And what is it doing? When is it finished? How do we know we can move on? The mosaic pixel picture won't go away on it's own. And what's the color wheel picture for? 

C) My log tells me:

2022-12-22T12:47:10.200Z [CALIBRATOR] (LutCalibrator.cpp:1217) ----------------- Preparing and saving LUT table --------------------
2022-12-22T12:47:10.200Z [CALIBRATOR] (LutCalibrator.cpp:1218) Initial mode: YES
2022-12-22T12:47:10.200Z [CALIBRATOR] (LutCalibrator.cpp:1219) Using YUV coefs: FCC
2022-12-22T12:47:10.200Z [CALIBRATOR] (LutCalibrator.cpp:1220) YUV table range: FULL
2022-12-22T12:47:10.317Z [CALIBRATOR] (LutCalibrator.cpp:1231) LUT RGB HDR table (1/3) is ready
2022-12-22T12:47:10.539Z [CALIBRATOR] (LutCalibrator.cpp:1269) LUT YUV HDR table (2/3) is ready
2022-12-22T12:47:11.614Z [CALIBRATOR] (LutCalibrator.cpp:1300) LUT YUV table (3/3) is ready
2022-12-22T12:47:12.313Z [CALIBRATOR] (LutCalibrator.cpp:1304) Your new LUT file is saved as /home/root/.hyperhdr/lut_lin_tables.3d. Is ready for usage: NO. It's temporary LUT table without HDR information..
2022-12-22T12:47:12.313Z [CALIBRATOR] (LutCalibrator.cpp:1305) ---------------------- LUT table is saved -----------------------
2022-12-22T12:47:12.314Z [CALIBRATOR] (LutCalibrator.cpp:1306) 
**2022-12-22T12:47:13.355Z [CALIBRATOR] Invalid CRC at: 0. CurrentColor: (29,45,102), Black: (48,137,53), White: (110,42,22), StartColor: (0,0,0), EndColor: (32,32,32).**

What am I doing wrong? HDR to SDR is disabled, Grabber Quality is 720p, Fullscreen Mode is enabled, Windows HDR is enabled and LG TV tells me it's HDR. 

And what to do when I have the finished LUT File? It's generated as "lut_lin_table" but I need "flat_lut_lin_table" for it to work. Do I have to rename it?

**Comment by NatyaSadella** [comment]
EDIT: I had to enable the Just Scan Setting on the LG TV to get it to work. With Just Scan on Automatic it cutted a bit of the taskbar, that's why it didn't work.

Log tells me calibration is now finished and I got a little message from HyperHDR that it's finished.

**Reply by awawa-dev** [reply]
You don't need to rename LUT file. This feature allows to have a separate LUT for a grabber and flatbuffers. In logs you can find out which 3d LUT is in use. There is no standard for full screen detection in JS so currently I implemented it only for Firefox.

**Reply by NatyaSadella** [reply]
A guy I know said I can do the calibration like this:

G2 TV (the one I use with ambilight) -> play a HDR file

then

CX TV that‘s connected to a HDR capable Windows PC -> Firefox HyperHDR -> Start Calibration 

Is this correct? I thought I have to use the mosaic pixel pattern on the TV I want to use.

**Reply by NatyaSadella** [reply]
And I need to rename the file to flat_lut_lin_tables after a calibration. Log tells me it found and loaded a LUT but it‘s never the one I have just created through calibration. The file is created at /home/root/.hyperhdr, Log finds and loads a flat_lut from another path, I believe /apps/services/ or something like that. If I rename the created LUT from lut lin to flat lut lin HyperHDR loads up the calibrated one.

**Reply by awawa-dev** [reply]
> And I need to rename the file to flat_lut_lin_tables after a calibration. Log tells me it found and loaded a LUT but it‘s never the one I have just created through calibration. The file is created at /home/root/.hyperhdr, Log finds and loads a flat_lut from another path, I believe /apps/services/ or something like that. If I rename the created LUT from lut lin to flat lut lin HyperHDR loads up the calibrated one.

HyperHDR comes without flat_lut_lin_table.3d so it's not base case scenario. Always control the log as I wrote.

**Reply by awawa-dev** [reply]
> A guy I know said I can do the calibration like this:
> 
> G2 TV (the one I use with ambilight) -> play a HDR file
> 
> then
> 
> CX TV that‘s connected to a HDR capable Windows PC -> Firefox HyperHDR -> Start Calibration
> 
> Is this correct? I thought I have to use the mosaic pixel pattern on the TV I want to use.

HyperHDR must be able to capture the mosaic. The calibration wont work when HyperHDR is run on webos.
